src/app/phast/losses/operations/co2-savings-phast/co2-savings-phast.component.ts

```diff
@@ -0,0 +1,362 @@
+      this.form.controls.eGridSubregion.patchValue(null);
+    }
+    this.calculate();
+  }
+
+  setBaselineSubregionForm() {
+    if (this.form.controls.eGridSubregion.value === null) {
+      // set the first from the subregion list as default
+      this.form.controls.eGridSubregion.patchValue(this.zipCodeSubRegionData[0]);
+    }
+    this.hasValidSubRegion = true;
+
+    this.setSubregionControlStatus();
+    if (!this.form.controls.userEnteredBaselineEmissions.value) {
+      this.setSubRegionEmissionsOutput();
+    }
+  }
+
+  setModificationSubregionForm() {
+    // set selected baseline subregion value
+    this.form.controls.eGridSubregion.patchValue(this.co2SavingsData.eGridSubregion);
+    this.hasValidSubRegion = true;
+    this.setSubregionControlStatus();
+
+    if (!this.form.controls.userEnteredModificationEmissions.value) {
+      this.setSubRegionEmissionsOutput();
+    }
+  }
+
+  setSubregionControlStatus() {
+    if (this.zipCodeSubRegionData.length === 0) {
+      // none found - form select is hidden, set form val to null
+      this.form.controls.eGridSubregion.disable();
+      this.form.controls.eGridSubregion.patchValue(null);
+    } else if (this.zipCodeSubRegionData.length === 1) {
+      // if only one result, no need to choose from select list
+      this.form.controls.eGridSubregion.disable();
+    } else if (this.isBaseline && !this.isFormDisabled) {
+      this.form.controls.eGridSubregion.enable();
+    }
+    this.cd.detectChanges();
+  }
+
+  setSubRegionEmissionsOutput() {
+    // setting default, so mod should not be user entered
+    if (!this.isBaseline) {
+      this.form.controls.userEnteredModificationEmissions.patchValue(false);
+    } else {
+      this.setUserEnteredModificationEmissions(false);
+    }
+
+    let subregionEmissions: SubregionEmissions = _.find(this.egridService.co2Emissions, (val) => { return this.form.controls.eGridSubregion.value === val.subregion; });
+    if (subregionEmissions) {
+      this.form.patchValue({
+        totalEmissionOutputRate: subregionEmissions.co2Emissions
+      });
+      this.calculate();
+    }
+  }
+
+  setLockedModificationValues() {
+    let modificationCo2SavingsData: Co2SavingsData = this.phastCO2SavingsService.modificationCo2SavingsData.getValue();
+    if (modificationCo2SavingsData) {
+      modificationCo2SavingsData.eGridSubregion = this.co2SavingsData.eGridSubregion;
+      modificationCo2SavingsData.zipcode = this.co2SavingsData.zipcode;
+      // IMPORTANT - the below vales were missing so bl values were not copied over to mod
+      modificationCo2SavingsData.energyType = this.co2SavingsData.energyType;
+      modificationCo2SavingsData.fuelType = this.co2SavingsData.fuelType;
+      modificationCo2SavingsData.energySource = this.co2SavingsData.energySource;
+      this.phastCO2SavingsService.modificationCo2SavingsData.next(modificationCo2SavingsData);
+    }
+  }
+
+  convertCO2SavingsData(co2SavingsData: Co2SavingsData): Co2SavingsData{
```

Comment:
I don't believe there's a scenario where the conversions in here will be executed. emission output rate is in the correct unit when the user enters it, and you're converting it when it comes from the csv also. I don't think you need this. Take a second look